src/test/rgw/test_rgw_url.cc

```diff
@@ -97,7 +97,7 @@ TEST(TestURL, InvalidHost)
     std::string user;
     std::string password;
     const std::string url = "http://exa_mple.com";
-    ASSERT_FALSE(parse_url_authority(url, host, user, password));
+    ASSERT_TRUE(parse_url_authority(url, host, user, password));
```

Comment:
interesting that it accepts this. the test name `InvalidHost` no longer makes sense for this, though. maybe we can find other ways to test boost::url's error handling?

Reply:
> interesting that it accepts this. the test name `InvalidHost` no longer makes sense for this, though. maybe we can find other ways to test boost::url's error handling?

do we want to test the boost:url thing?
or options that users can set as part of URL and which could be valid url ?
also checking on whether `_` allowed in url, it is allowed